### DISCUSSION TITLE
Drop support for older RHEL like distributions (6-)

### DISCUSSION THREAD

**Discussion by StykMartin** [restraint-harness/restraint#207] [Ideas]
Hello,

I would like to open discussion here.
I believe that we should drop support for RHEL6-.
This will help us to keep up with upstream and merge some important patches like Glib bump in #197

**Comment by JohnVillalovos** [comment]
We would be fine with this as we only use RHEL 8 (or higher whenever that happens).

And staying closer to upstream will be a big benefit.

**Comment by renanrodrigo** [comment]
I do think we would be fine as well, as `0.3.2` is doing good enough for the tests on <6. We informed people about it and there was no major pushback AFAIK.

**Comment by StykMartin** [comment]
I believe we can push this change to `0.4.0`. And work on any outstanding issues to finish the `0.3.z` life cycle.

**Comment by AloisMahdal** [comment]
We're relying on restraint to help with upgrades and migrations:

 * CentOS-6 -> RHEL-6,
 * RHEL-6 -> RHEL-7.

Sure, there's an expiration date on this testing effort, but I've seen old buried distros crawl back from the graves, so I'd like to know what @shaded-enmity, @Rezney or @bocekm think about the timeline.  Guys, is there a date when we can be absolutely sure there won't be any testing needed on RHEL-6/CentOS-6?

**Reply by StykMartin** [reply]
Well. The question is if there is any outstanding feature that is missing. I believe we covered almost everything what would you need for your testing. In worst case we can still backport fix to 0.3.z

**Reply by AloisMahdal** [reply]
On RHEL-6 we've actually only switched to restraint recently(*) and I haven't had the opportunity to convince myself 100%, but I think current 0.3.2 works fine for us.

So I guess it's not worth holding you back.

(* there has been some strange AVC issue so we kept stuck with beah -- but we recently stopped raising AVC's completely so that allowed us to move our rhel6 tests to restraint as well, yay!)

**Reply by StykMartin** [reply]
So, we are planning to do another Community meeting where we can discuss this topic.
If you would like to, here is link https://github.com/beaker-project/beaker-project.org/wiki/Community-Meeting-%2302-(2021-04-23)

I will update it soon with rest of the details (like url and stuff)

**Reply by bocekm** [reply]
Regarding conversions from CentOS 6 to RHEL 6, there's no indication as of now that we would stop allowing them before the end of the Extended life cycle support (June 30, 2024).
Even though CentOS 6 to RHEL 6 conversions are not officially supported, we'll continue adding new features and fixing bugs for that conversion path as well, which will require verification.
Yet, in close future, we will not 100% rely on Beaker in terms of testing the conversions as we'll be utilizing other system provisioners as well. So, all in all, dropping CentOS 6 from Beaker will have close to zero impact on our testing.

**Reply by AloisMahdal** [reply]
Sounds good.  As Martin has clarified, this is not even about dropping CentOS from Beaker, just stopping new restraint fixes/features from appearing on rhel6-likes.  That said the centos6 images in our beaker instance are currently broken anyway so all that would remain would be the RHEL-6 > RHEL-7 conversions, but then again, `restraint-0.3.2` works just fine...
.. so it's no big deal.

**Reply by StykMartin** [reply]
Right.
Beaker is not affected by this change. Only Restraint is affected. We will stop making new releases for Restraint and it will stay on the latest `0.3.z`. Further Restraint releases (`0.y.z`) will be for RHEL 7/8/9 + Fedora as always.

It is possible that I will propose to drop RHEL6- from Beaker as well. But I'm still gathering feedback. However, Red Hat will be not affected by this change anyway, **GitHub is no longer source of truth for Red Hat Beaker instance.** Based on the latest information, you are running a locked 28.2 version w/ epoch 1. When Red Hat is ready, you may backport new stuff that we are planning to introduce to Beaker.

**Reply by renanrodrigo** [reply]
Accurate. We did not even bumped the epoch, but every new minor/release for 28 will be done in house, and upstream is not a source of truth for our Beaker instance anymore.

**Comment by StykMartin** [comment]
I believe we all agree to drop RHEL6- support for Restraint.
This message will be shared at the community meeting and we will make this change effective from the `0.4.0` release.
